cluster/ubuntu/util.sh

```diff
@@ -107,7 +109,7 @@ function trap-add {
   trap "${handler}" ${signal}
```

Comment:
Oh!  Double quotes is correct here.  You want the trap to use the current (and local!) expansion of handler.